spec/rest/defaults.test.js

```diff
@@ -1,258 +1,245 @@
+			expect(normalisedOptions.tlsPort).to.equal(443);
+			expect(normalisedOptions.fallbackHosts.sort()).to.deep.equal(Defaults.FALLBACK_HOSTS.sort());
+			expect(normalisedOptions.tls).to.equal(true);
+
+			expect(Defaults.getHosts(normalisedOptions).length).to.deep.equal(4);
+			expect(Defaults.getHosts(normalisedOptions)[0]).to.deep.equal(normalisedOptions.restHost);
+			expect(Defaults.getHost(normalisedOptions, 'rest.ably.io', false)).to.deep.equal('rest.ably.io');
+			expect(Defaults.getHost(normalisedOptions, 'rest.ably.io', true)).to.deep.equal('realtime.ably.io');
+
+			expect(Defaults.getPort(normalisedOptions)).to.equal(443);
+			counter.assert();
+			done();
+		});
+
+		/* init with production environment */
+		it('defaults_production', function(done) {
+			counter.expect(11);
+			var normalisedOptions = Defaults.normaliseOptions({environment: 'production'});
+
+			expect(normalisedOptions.restHost).to.equal('rest.ably.io');
+			expect(normalisedOptions.realtimeHost).to.equal('realtime.ably.io');
+			expect(normalisedOptions.port).to.equal(80);
+			expect(normalisedOptions.tlsPort).to.equal(443);
+			expect(normalisedOptions.fallbackHosts.sort()).to.deep.equal(Defaults.FALLBACK_HOSTS.sort());
+			expect(normalisedOptions.tls).to.equal(true);
+
+			expect(Defaults.getHosts(normalisedOptions).length).to.deep.equal(4);
+			expect(Defaults.getHosts(normalisedOptions)[0]).to.deep.equal(normalisedOptions.restHost);
+			expect(Defaults.getHost(normalisedOptions, 'rest.ably.io', false)).to.deep.equal('rest.ably.io');
+			expect(Defaults.getHost(normalisedOptions, 'rest.ably.io', true)).to.deep.equal('realtime.ably.io');
+
+			expect(Defaults.getPort(normalisedOptions)).to.equal(443);
+			counter.assert();
+			done();
+		});
+
+		/* init with given environment */
+		it('defaults_given_environment', function(done) {
+			counter.expect(11);
+			var normalisedOptions = Defaults.normaliseOptions({environment: 'sandbox'});
+
+			expect(normalisedOptions.restHost).to.equal('sandbox-rest.ably.io');
+			expect(normalisedOptions.realtimeHost).to.equal('sandbox-realtime.ably.io');
+			expect(normalisedOptions.port).to.equal(80);
+			expect(normalisedOptions.tlsPort).to.equal(443);
+			expect(normalisedOptions.fallbackHosts.sort()).to.deep.equal(Defaults.environmentFallbackHosts('sandbox').sort());
+			expect(normalisedOptions.tls).to.equal(true);
+
+			expect(Defaults.getHosts(normalisedOptions).length).to.deep.equal(4);
+			expect(Defaults.getHosts(normalisedOptions)[0]).to.deep.equal(normalisedOptions.restHost);
+			expect(Defaults.getHost(normalisedOptions, 'sandbox-rest.ably.io', false)).to.deep.equal('sandbox-rest.ably.io');
+			expect(Defaults.getHost(normalisedOptions, 'sandbox-rest.ably.io', true)).to.deep.equal('sandbox-realtime.ably.io');
+
+			expect(Defaults.getPort(normalisedOptions)).to.equal(443);
+			counter.assert();
+			done();
+		});
+
+		/* init with given environment */
+		it('defaults_given_environment2', function(done) {
```

Comment:
Let's call this `defaults_given_environment_and_fallbackHostsUseDefault` but keep the existing comment:

```
/* init with given environment and default fallbacks */
/* will emit a deprecation warning */
```